microscopium/bokeh_app.py

```diff
@@ -7,9 +7,10 @@
 from bokeh.application.handlers.function import FunctionHandler
 from bokeh.plotting import figure, ColumnDataSource
 from bokeh.layouts import row
-from bokeh.models.tools import (ResetTool, PanTool, WheelZoomTool, TapTool,
-                                BoxSelectTool, PolySelectTool, UndoTool,
-                                RedoTool, HoverTool)
+from bokeh.models.tools import (HoverTool, TapTool, PanTool,
+                                WheelZoomTool, BoxZoomTool,
+                                BoxSelectTool, PolySelectTool, LassoSelectTool,
```

Comment:
Incidentally, my experience with LassoSelect is that it is way too slow, which is why I left it out. I don't feel strongly about it being left out, just letting you know the reason. =)

Reply:
Yeah, I assumed it'd be pretty slow - does polyselect handle odd shapes much faster?

Reply:
In my experiments of a bajillion years ago, it did. I don't know how things might have changed since then. I imagine Lasso is probably implemented as a polygon with single-pixel-wide sides, so you'd expect it to be slower.

Reply:
I can't seem to figure out how to get the PolySelect tool to, well, actually select things. Is it just kind of unintuitive, or is that likely a sign of a bigger problem?

Reply:
Apparently you need to double-click to finish the selection

Reply:
Oh, just unintuitive. I think I need to both start and end with a double click. And it's not really showing me the partially completed polygon shape, either.

Reply:
Maybe we should just stick to rectangles for now? ;)

Reply:
If you like, I'm not too fussed either way.